### PR TITLE
Remove most references to devlopment channels in the entity URL.

### PR DESCRIPTION
**Pull request by mhilton**
There are some references that have been kept to ensure that if a
development URL was used with the charmstore, it could not cause a
violation of invariants in the database.

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1539/
Test PASSed.

File: internal/charmstore/addentity.go [hunk 1]
```diff
@@ -412,7 +412,7 @@ func (s *Store) addCharm(c charm.Charm, p addParams) (err error) {
 	// final gateway before a potentially invalid url might be stored
 	// in the database.
 	id := p.url.URL
-	logger.Infof("add charm url %s; prev %d; dev %v", &id, p.url.PromulgatedRevision, p.url.Development)
+	logger.Infof("add charm url %s; prev %d", &id, p.url.PromulgatedRevision)
```
Comment (frankban) [inline internal/charmstore/addentity.go:415]: s/prev/promulgated rev/ ? as I always read that as "previous" and it's confusing [shrug]

File: internal/charmstore/addentity.go [hunk 2]
```diff
@@ -515,13 +513,12 @@ func (s *Store) addEntity(entity *mongodoc.Entity) (err error) {
 		Write: perms,
 	}
 	baseEntity := &mongodoc.BaseEntity{
-		URL:             entity.BaseURL,
-		User:            entity.User,
-		Name:            entity.Name,
-		Public:          false,
-		ACLs:            acls,
-		DevelopmentACLs: acls,
```
Comment (frankban) [inline internal/charmstore/addentity.go:523]: I think we will reintroduce DevelopmentACLs later, is it worth removing them now?

**Comment by frankban** [comment]
:+1: with a couple of minor suggestions.

File: internal/charmstore/addentity.go (hunk 2 above)
Comment (mhilton) [inline internal/charmstore/addentity.go:523]: I've only removed filling them out, they are still in the document. If I read the design correctly they will be modified slightly anyway.

File: internal/charmstore/addentity.go (hunk 1 above)
Comment (mhilton) [inline internal/charmstore/addentity.go:415]: I see your point, I'll address it in the next PR

**Comment by mhilton** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Pull request merged**